### PR TITLE
jbide-22626: Updating deployed docker image on OpenShift does not work

### PR DESCRIPTION
**Pull request by bdshadow**
@jeffmaury @jcantrill @adietish , please, have a look
Possibly it is better to do update in a separate method, but then we need to get DC from connection again. Also i believe, that some better naming for the method can be created

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java [hunk 1]
```diff
@@ -124,18 +124,26 @@ protected IStatus doRun(IProgressMonitor monitor) {
 		return Status.OK_STATUS;
 	}
 
-	private boolean isUpdate(Connection connection, String project, String name) {
+	private boolean isUpdateThenDo(Connection connection, String project, String name) {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java]: isUpdate then do what? I agree that isUpdate() isnt a good choice for a method name, but I still dont get (by reading the method signature) what this method does.
Is this something like "updateTriggerIfIsUpdate" or similar?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java [hunk 2]
```diff
@@ -124,18 +124,26 @@ protected IStatus doRun(IProgressMonitor monitor) {
...
-			return trigger != null && 
-				ResourceKind.IMAGE_STREAM_TAG.equals(trigger.getKind()) && 
-				StringUtils.isNotBlank(trigger.getNamespace()) &&  
-				connection.getResource(ResourceKind.IMAGE_STREAM, trigger.getNamespace(), trigger.getFrom().getName()) != null;
-		}catch(NotFoundException e) {
+			if (trigger == null || 
+				!ResourceKind.IMAGE_STREAM_TAG.equals(trigger.getKind()) || 
+				StringUtils.isBlank(trigger.getNamespace()) ||  
+				connection.getResource(ResourceKind.IMAGE_STREAM, trigger.getNamespace(), trigger.getFrom().getName()) == null) {
+				return false;
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java]: could we please have unit-tests to assert these rather complex alternative conditions?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizard.java [hunk 3]
```diff
@@ -72,11 +71,8 @@ public DeployImageWizard(IDockerImage image, Connection connection, IProject pro
 	private String getImageNameWithTag(IDockerImage image) {
 		String imageName = image.repo();
 		if(!image.tags().isEmpty()) {
-			List<String> tags = new ArrayList<>(image.tags());
-			if(tags.size() > 1) {
-				Collections.sort(tags); //same as in docker explorer
-			}
-			imageName += ":" + tags.get(0);
+			TreeSet<String> tags = new TreeSet<>(image.tags());
+			imageName += ":" + tags.first();
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizard.java]: good catch!

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java (hunk 1 above)
Comment (bdshadow) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java]: is Update? then do update =) That's what i meant. It was isUpdate before and added update to the same method for not to get resources from connection 2 times.
I agree, that  `updateTriggerIfUpdate` is better

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java (hunk 2 above)
Comment (bdshadow) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java]: ok, i'll work on it

**Comment by bdshadow** [comment]
I had some mess in branches locally, now fixed it. Added test on checking tag updates.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java (hunk 2 above)
Comment (bdshadow) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java]: added

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
+1, works for me.

**Pull request merged**